main.tf

```diff
@@ -1,65 +1,88 @@
+  count = local.create_vpc_ipam && var.vpc_ipam.ipam != null ? 1 : 0
 
-module "vpc_ipam_pool" {
-  source                            = "./modules/pool"
-  address_family                    = var.pool.address_family
-  allocation_default_netmask_length = var.pool.allocation_default_netmask_length
-  allocation_max_netmask_length     = var.pool.allocation_max_netmask_length
-  allocation_min_netmask_length     = var.pool.allocation_min_netmask_length
-  allocation_resource_tags          = var.pool.allocation_resource_tags
-  auto_import                       = var.pool.auto_import
-  aws_service                       = var.pool.aws_service
-  ipam_scope_id                     = var.pool.ipam_scope_id
-  locale                            = var.pool.locale
-  publicly_advertisable             = var.pool.publicly_advertisable
-  public_ip_source                  = var.pool.public_ip_source
-  source_ipam_pool_id               = var.pool.source_ipam_pool_id
-  tags                              = var.pool.tags
+  description = var.vpc_ipam.ipam.description
+  dynamic "operating_regions" {
+    for_each = var.vpc_ipam.ipam.operating_regions
+    content {
+      region_name = operating_regions.value.region_name
+    }
+  }
+  tags    = var.vpc_ipam.ipam.tags
+  cascade = var.vpc_ipam.ipam.cascade
 }
 
-module "vpc_ipam_pool_cidr" {
-  source                               = "./modules/pool-cidr"
-  cidr                                 = var.pool_cidr.cidr
-  cidr_authorization_context_message   = var.pool_cidr.cidr_authorization_context_message
-  cidr_authorization_context_signature = var.pool_cidr.cidr_authorization_context_signature
-  ipam_pool_id                         = var.pool_cidr.ipam_pool_id
-  netmask_length                       = var.pool_cidr.netmask_length
+# #plural, and nestable to 10 deep
+resource "aws_vpc_ipam_pool" "this" {
+  for_each = { for pool in var.vpc_ipam.pools : pool.name => pool }
+
+  address_family                    = try(lower(each.value.address_family), null)
+  allocation_default_netmask_length = each.value.allocation_default_netmask_length
+  allocation_max_netmask_length     = each.value.allocation_max_netmask_length
+  allocation_min_netmask_length     = each.value.allocation_min_netmask_length
+  allocation_resource_tags          = each.value.allocation_resource_tags
+  auto_import                       = each.value.auto_import
+  aws_service                       = try(lower(each.value.aws_service), null)
+  description                       = each.value.description
+  ipam_scope_id                     = each.value.ipam_scope_id
+  #could be output of aws_vpc_ipam or aws_vpc_ipam_scope - try statement?
+  locale                = try(lower(each.value.locale), null)
+  publicly_advertisable = each.value.publicly_advertisable
+  public_ip_source      = each.value.public_ip_source
+  source_ipam_pool_id   = each.value.source_ipam_pool_id
+  tags                  = each.value.tags
 }
```

Comment:
looked at the tgw repo and the try() statements are def helpful...

this resource and corresponding `var` can technically be plural, but pools can parent, Nth child, and Nth grandchild. Having a for loop to go through each one is not great. `vpc_ipam.pools.source_ipam_pool_id` delinates a child relationship if specific, but nothing delinates sibling relationship (other than having same parent) 

Just caught up a little with the `vpc_ipam.pools` being a list of objects, and not knowing what the `source_ipam_pool_id` will be yet

Reply:
I don't think we can support a hierarchy of pools easily. The user will have to build up the structure for that themselves, calling the module over and over, chaining outputs together. It is still good for this to be a list, because the user can create one pool in one module call, then use that output as the input to create two child pools in a second module call.